opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -78,12 +78,14 @@
    * @param format the Format of the carrier
    * @param carrier the carrier for the SpanContext state. All Tracer.extract() implementations must support
    *                io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
-   * @returns the SpanContext instance extracted from the carrier
+   *
+   * @return the SpanBuilder instance holding context to create a Span.
+   * A noop builder is returned if the carrier held no trace information.
    *
    * @see io.opentracing.propagation.Format
    * @see io.opentracing.propagation.Format.Builtin
    */
-  <C> SpanContext extract(Format<C> format, C carrier);
+  <C> SpanBuilder extract(Format<C> format, C carrier);
```

Comment:
@michaelsembwever I'm not in favor of this change... all other languages inject and extract SpanContexts... this makes things like multiple-parent specification uniform whether SpanContexts came from the wire or within the process, etc.